### PR TITLE
Update Dart webassembly-language-support.md

### PR DESCRIPTION
**Pull request by KyleFin**
Update language matrix to be aligned with @kevmoo's https://github.com/fermyon/developer/commit/6cb4c1e4481c423dca3bc7d3c1fc7f4b952d2949

- Dart (and Flutter) have core and browser support: https://dart.dev/web/wasm
- WASI support is tracked in https://github.com/dart-lang/sdk/issues/56366

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title`, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep $'\r' | wc -l` and expect 0 as a result)
- [ ] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [ ] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors
- [ ] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

File: content/wasm-languages/webassembly-language-support.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Some languages, like CSS, PowerShell, and "Shell", don't really have a meaningfu
 | [Kotlin (JVM)][Kotlin]    | ✅    | ✅      | ✅   | ⏳       |
 | [Kotlin (Wasm)][Kotlin]   | ⏳    | ✅      | ✅   | ❌       |
 | [Rust][Rust]              | ✅    | ✅      | ✅   | ✅       |
-| [Dart][Dart]              | ❌    | ⏳      | ❌   | ❌       |
+| [Dart][Dart]              | ✅    | ✅      | ⏳   | ❌       |
 
 * _Core_ means there is an implementation of WebAssembly 1.0
```
Comment (KyleFin) [inline content/wasm-languages/webassembly-language-support.md:44]: I'm curious about this definition. GC languages require 2.0, right? So should `Core` column be ❌ for Dart, Java, Kotlin, etc?

If not, then I believe Dart should be ✅

**Review by itowlson — APPROVED** [review]

File: content/wasm-languages/webassembly-language-support.md (hunk 1 above)
Comment (itowlson) [reply content/wasm-languages/webassembly-language-support.md:44]: @technosophos I'm going to merge this on the basis of Kotlin being green, but I may be misunderstanding your intent with the Core column.  If so let me know and we can put it back to what you intended (running it past Kyle of course).

**Comment by itowlson** [comment]
CI fail is an unrelated broken link at sigstore

**Comment by itowlson** [comment]
My apologies, @KyleFin, I forgot to say thank you for this!

**Comment by KyleFin** [comment]
Happy to help! Thanks @itowlson :)

**Comment by iapicca** [comment]
I see Dart completely disappeared from the [languages support ](https://developer.fermyon.com/spin/v3/language-support-overview)

cc @kevmoo

**Comment by KyleFin** [comment]
@iapicca, I think those pages are about Spin support, right? I don't know that Spin has Dart support.

My original update was about the WebAssembly language matrix https://developer.fermyon.com/wasm-languages/webassembly-language-support

**Comment by itowlson** [comment]
This is correct.  You can see from the last column in the "what languages support WebAssembly" matrix (which @KyleFin updated) that there is no Spin SDK for Dart (I think GC could currently be a blocker for this).  Hence no coverage in the Spin docs.  Sorry for the confusion @iapicca

**Pull request merged**